### PR TITLE
[CIS-543] Document picker and attachment sending

### PR DESCRIPTION
**Pull request by mantergo**
**In this PR:** 

- Implement image and document attachments sending

### CODE REVIEW

File: Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerViewController.swift [hunk 1]
```diff
@@ -60,12 +60,19 @@ open class MessageComposerViewController<ExtraData: ExtraDataTypes>: ViewControl
...
+        picker.mediaTypes = ["public.image"]
         picker.sourceType = .photoLibrary
         picker.delegate = self
         return picker
     }()
     
+    public private(set) lazy var documentPicker: UIDocumentPickerViewController = {
+        let picker = UIDocumentPickerViewController(documentTypes: ["public.item"], in: .import)
+        picker.delegate = self
+        picker.allowsMultipleSelection = false
```
Comment (VojtaStavik) [inline Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerViewController.swift:72]: Why? :)

File: Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerViewController.swift [hunk 2]
```diff
@@ -207,15 +214,42 @@ open class MessageComposerViewController<ExtraData: ExtraDataTypes>: ViewControl
...
+            
+            actionSheet.addAction(UIAlertAction(title: "Photo", style: .default, handler: { [weak self] _ in
+                guard let self = self else { return }
+                self.present(self.imagePicker, animated: true, completion: nil)
+            }))
+            
+            actionSheet.addAction(UIAlertAction(title: "Cancel", style: .cancel))
+            
+            return actionSheet
+        }
```
Comment (VojtaStavik) [inline Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerViewController.swift:242]: nit: I know this is just a temp solution but we should still use the string file for this

File: Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerViewController.swift [hunk 3]
```diff
@@ -251,18 +285,70 @@ open class MessageComposerViewController<ExtraData: ExtraDataTypes>: ViewControl
...
         }
     }
     
+    var documentAttachments: [AttachmentInfo] = [] {
+        didSet {
+            didUpdateDocumentAttachments()
+        }
+    }
+    
+    var attachmentSeeds: [_ChatMessageAttachment<ExtraData>.Seed] {
```
Comment (VojtaStavik) [inline Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerViewController.swift:314]: nit: All these things should be `open`

File: Sources_v3/StreamChatUI/UIConfig.swift [hunk 4]
```diff
@@ -261,6 +261,27 @@ public extension UIConfig {
...
+            "pptx": UIImage(named: "pptx", in: .streamChatUI)!,
+            "rar": UIImage(named: "rar", in: .streamChatUI)!,
+            "rtf": UIImage(named: "rtf", in: .streamChatUI)!,
+            "tar.gz": UIImage(named: "tar.gz", in: .streamChatUI)!,
+            "txt": UIImage(named: "txt", in: .streamChatUI)!,
+            "xls": UIImage(named: "xls", in: .streamChatUI)!,
+            "xlsx": UIImage(named: "xlsx", in: .streamChatUI)!,
+            "zip": UIImage(named: "zip", in: .streamChatUI)!
+        ]
+        public var fallbackDocumentPreview: UIImage = UIImage(named: "generic", in: .streamChatUI)!
```
Comment (VojtaStavik) [inline Sources_v3/StreamChatUI/UIConfig.swift:284]: 👍

**Review by VojtaStavik — APPROVED** [review]
Looks good to me 🚢 🇮🇹

File: Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerViewController.swift (hunk 1 above)
Comment (mantergo) [reply Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerViewController.swift:72]: Good question 🤷‍♂️

**Pull request merged**